test/net/sourceforge/kolmafia/textui/command/MirrorCommandTest.java

```diff
@@ -0,0 +1,176 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static internal.helpers.Player.withSkill;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import internal.helpers.Cleanups;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.utilities.FileUtilities;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class MirrorCommandTest extends AbstractCommandTestBase {
+
+  public MirrorCommandTest() {
+    this.command = "mirror";
+  }
+
+  @BeforeEach
+  public void stopMirrorLogging() {
+    execute("");
+  }
+
+  private String getMirrorLog(String fileName) {
+    StringBuilder builder = new StringBuilder();
+    try (BufferedReader reader =
+        FileUtilities.getReader(new File(KoLConstants.ROOT_LOCATION, "chats/" + fileName))) {
+      String line;
+      // Read the full file to a string
+      while ((line = FileUtilities.readLine(reader)) != null) {
+        if (builder.length() > 0) {
+          builder.append("\n");
+        }
+        builder.append(line);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return builder.toString();
+  }
+
+  @Test
+  public void testMirrorWritesHTML() {
+    // Open the mirror
+    execute("chats/test_writes_html.txt");
+    // When `> ` is used as a prefix, the RequestLogger will colorize it assuming its a command
+    RequestLogger.printLine("> Fake command input");
+    // Log another line, this should be raw.
+    RequestLogger.printLine("Raw Line");
+    // Close the mirror
+    execute("");
+    String mirrorOutput = getMirrorLog("test_writes_html.txt");
+    assertThat(mirrorOutput, containsString("<font color=olive>> Fake command input</font><br>"));
+    assertThat(mirrorOutput, containsString("Raw Line"));
+  }
+
+  @Test
+  public void testMirrorDoesNotAddHTML() {
+    // Open the mirror
+    execute("chats/test_writes.txt");
+    // Log a raw line, this should be raw.
+    RequestLogger.printLine("Raw Line");
+    // Close the mirror
+    execute("");
+    String mirrorOutput = getMirrorLog("test_writes.txt");
+    assertThat(mirrorOutput, containsString("Raw Line"));
```

Comment:
This test does not actually check for the existence of HTML. Only that the string contains the text we're looking for.

Either check the whole file contents match as expected, or perhaps check for enclosing tags < and >

Reply:
Noted and tweaked.